src/Chapter2.hs

```diff
@@ -858,8 +890,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+rewind :: [a] -> [a]
+rewind = rw []
+    where
+        rw ::  [a] -> [a] -> [a]
+        rw y [] = y
+        rw y (x:xs) = rw (x:y) xs
```

Comment:
Amazing solutions! :1st_place_medal: 

Usually, such helper function is called `go` :slightly_smiling_face: You can read more about this pattern in our blog: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go